hdf5-types/src/dyn_value.rs

```diff
@@ -527,12 +534,14 @@ unsafe impl DynClone for DynVarLenString<'_> {
     fn dyn_clone(&mut self, out: &mut [u8]) {
         debug_assert_eq!(out.len(), mem::size_of::<usize>());
         if !self.get_ptr().is_null() {
+            debug_assert!(is_aligned_for_type::<DynVarLenString>(out.as_ptr()));
             unsafe {
                 let raw_len = self.raw_len();
-                let dst = crate::malloc(raw_len + 1) as *mut _;
+                let dst = crate::malloc(raw_len + 1).cast();
                 ptr::copy_nonoverlapping(self.get_ptr(), dst, raw_len);
-                *dst.add(raw_len) = 0;
-                *(out.as_mut_ptr() as *mut *const u8) = dst as _;
+                dst.add(raw_len).write(0);
+                let outptr = out.as_mut_ptr().cast::<*const u8>(); // alignment asserted above
```

Comment:
Hm, not sure about this. How can a `*const u8` be non-aligned?

What's the story about the align checks in this pr?

Reply:
This pointer is an `*mut *const u8` which needs 4/8-bytes of alignment